Makefile

```diff
@@ -9,10 +9,8 @@ PROJ = google_containers
 
 PKG_SOURCES := $(shell find pkg -name '*.go')
 
-node-problem-detector: ./bin/node-problem-detector
-
-./bin/node-problem-detector: $(PKG_SOURCES) node_problem_detector.go
-	CGO_ENABLED=0 GOOS=linux godep go build -a -installsuffix cgo -ldflags '-w' -o node-problem-detector
+node-problem-detector: $(PKG_SOURCES) node_problem_detector.go
+	GOOS=linux go build -ldflags '-w -extldflags "-static"' -o node-problem-detector
```

Comment:
@Random-Liu 
After some search, i think i can make conclusions:
* the way i add `-extldflags "-static"` to `go build` is not necessary and wrong. After adding this option, node-problem-detector can be compiled to be a static linked binary. However, this will make it panic just like what you have posted in [#discussion_r93560095](https://github.com/kubernetes/node-problem-detector/pull/54#discussion_r93560095).
* we should compile node-problem-detector in dynamic only.